### PR TITLE
Add JooqMock and ApiDto updates from project implementations

### PR DESCRIPTION
**Pull request by connernilsen**
- Add `ApiDto`
- Update `JooqMock`

### CODE REVIEW

File: service/src/test/java/com.codeforcommunity/JooqMock.java [hunk 1]
```diff
@@ -189,14 +195,31 @@ int getCallCount() {
...
-   * default.
-   */
+  /** Constructor for JooqMock. 'UNKNOWN' and 'DROP/CREATE' operations are added by default. */
   public JooqMock() {
     // create DSL context
     MockConnection connection = new MockConnection(this);
-    context = DSL.using(connection, SQLDialect.POSTGRES);
+    raw_context = DSL.using(connection, SQLDialect.POSTGRES);
+    context = spy(raw_context);
+    id = 1;
```
Comment (jackblanc) [inline service/src/test/java/com.codeforcommunity/JooqMock.java:204]: Magic number?

File: service/test/AuthorizationProcessorImplTest.java [hunk 2]
```diff
@@ -1,176 +0,0 @@
-import com.auth0.jwt.JWT;
```
Comment (jackblanc) [inline service/test/AuthorizationProcessorImplTest.java:1]: you sure we want to delete this whole spec?

File: service/src/test/java/com.codeforcommunity/JooqMock.java (hunk 1 above)
Comment (connernilsen) [reply service/src/test/java/com.codeforcommunity/JooqMock.java:204]: Its initializing the variable. It gets updated (it's not thread safe but we don't really multi-thread a lot so it's fine), so it doesn't stay the same. It's used to set the 'current' ID of an object on insert, and it gets incremented after every insert.

File: service/test/AuthorizationProcessorImplTest.java (hunk 2 above)
Comment (connernilsen) [reply service/test/AuthorizationProcessorImplTest.java:1]: Yeah, we got rid of it in the other repos, and it's in the wrong location to even run. Do you want me to add it  back in and move it to where it should be?

Comment (jackblanc) [reply service/test/AuthorizationProcessorImplTest.java:1]: If you think it's easier then yeah. Up to you.

Comment (connernilsen) [reply service/test/AuthorizationProcessorImplTest.java:1]: I'll add it back in if it still runs. Last I remember there were a lot of changes, but we'll see.

Comment (connernilsen) [reply service/test/AuthorizationProcessorImplTest.java:1]: Yeah, it looks like things are completely different now.

**Pull request merged**